src/data/nat/prime.lean

```diff
@@ -1022,4 +1045,43 @@ begin
   rw [mem_factors_mul_of_pos ha hb p, list.mem_union]
 end
 
+
+open list
+
+/-- If `p` is a prime factor of `a` then `p` is also a prime factor of `a * b` for any `b > 0` -/
+lemma mem_factors_mul_left {p a b : ℕ} (hpa : p ∈ a.factors) (hb : 0 < b) : p ∈ (a*b).factors :=
+begin
+  rcases a.eq_zero_or_pos with rfl | ha,
+  { rw nat.factors_zero at hpa, exact absurd hpa (not_mem_nil p) },
+  { rw mem_factors ha at hpa,
+    exact (mem_factors (mul_pos ha hb)).mpr ⟨hpa.1, dvd_mul_of_dvd_left hpa.2 b⟩ }
+end
+
+/-- If `p` is a prime factor of `b` then `p` is also a prime factor of `a * b` for any `a > 0` -/
+lemma mem_factors_mul_right {p a b : ℕ} (hpb : p ∈ b.factors) (ha : 0 < a) : p ∈ (a*b).factors :=
+by { rw mul_comm, exact mem_factors_mul_left hpb ha }
+
+/-- If `p` is a prime factor of `a` then the power of `p` in `a` (i.e. the greatest `k` such that
+`p^k` divides `a`) is the same as the power of `p` in `a * b`, for any `b` coprime to `a`. -/
+lemma factors_count_eq_of_coprime_left {p a b : ℕ} (hab : coprime a b) (hpa : p ∈ a.factors) :
+  list.count p a.factors = list.count p (a * b).factors :=
+begin
+  rw [perm.count_eq (perm_factors_mul_of_coprime hab) p, count_append],
+  simpa only [count_eq_zero_of_not_mem (coprime_factors_disjoint hab hpa)],
+end
+
+/-- If `p` is a prime factor of `b` then the power of `p` in `b` (i.e. the greatest `k` such that
+`p^k` divides `b`) is the same as the power of `p` in `a * b`, for any `a` coprime to `b`. -/
+lemma factors_count_eq_of_coprime_right {p a b : ℕ} (hab : coprime a b) (hpb : p ∈ b.factors) :
+  list.count p b.factors = list.count p (a * b).factors :=
+by { rw mul_comm, exact factors_count_eq_of_coprime_left (coprime_comm.mp hab) hpb }
+
+lemma le_factors_count_mul {p a b : ℕ} (ha : 0 < a) :
```

Comment:
```suggestion
lemma le_factors_count_mul_right {p a b : ℕ} (ha : 0 < a) :
```
The name I gave was bad, sorry.

Also the `_left` version of this would be useful.

Lastly, (the left version of) this lemma can simplify the proof of `mem_factors_mul_left` using `list.count_pos`.